protobuf/coreapi/api.proto

```diff
@@ -157,9 +157,23 @@ message ListenResultRequest {
 message ResultData {
   string executionID = 1;             // The unique identifier of the execution.
   string taskKey = 2;                 // The key of the executed task.
-  string outputKey = 3;               // The output's key from the returned task.
-  string outputData = 4;              // The output's data from the returned task, encoded in JSON.
   repeated string executionTags = 5;  // The list of tags associated with the execution
+  oneof result {
```

Comment:
I'm a bit worried on this one, I just want to make sure that this is not breaking for the existing libraries (it's ok if that breaks when there is an error but not when there is a success.

We should make sure that libraries are always compatible with that or at least update the libraries and make them as breaking release if needed.

Reply:
I tested manually and it works. We should have integration tests to be sure https://forum.mesg.com/t/automatize-manual-testing-for-core-true-integration-testing/94/7

Reply:
Why not having something like that
```proto
message ResultData {
  string executionID = 1;
  string taskKey = 2;
  string outputKey = 3;
  repeated string executionTags = 5;  
  
  oneof result {
    string outputData = 4;
    Error error = 7;
  }

  message Error {
    string message = 1;
  }
}
```

We can actually provide the `outputKey` even when there is an error so let's give more context and like that we don't need to deprecate or change anything on the api, just add the error fields.

Reply:
two catches here:

* output key might be invalid send by the client. (not defined in the yml)
* output data might be a bad json so it cannot be unmarshaled. therefore we'll not be able to set it _execution.OutputData_ which is not convenient.

Reply:
outputKey can be invalid the point here is that it will always be here (even if empty). If invalid we will have the error as result.

For the second part I don't really understand, it's either we can parse it and we have the outputData or we cannot parse it and we get the error but i'm not sure I understand what you say

Reply:
what i mean is, even if we can provide the output key when there is an error we never can do it for the output data. because we cannot parse it to a map. because of that we shouldn't set both of them. invalid key and wrong formatted output data can be put inside the error message itself.